server/src/main/java/org/opensearch/index/shard/CheckpointRefreshListener.java

```diff
@@ -43,7 +43,14 @@ protected boolean performAfterRefreshWithPermit(boolean didRefresh) {
         if (didRefresh
             && shard.state() == IndexShardState.STARTED
             && shard.getReplicationTracker().isPrimaryMode()
-            && !shard.indexSettings.isSegRepWithRemoteEnabled()) {
+            && !shard.indexSettings.isSegRepWithRemoteEnabled()
+            /*
+             During remote store migration, the isSegRepWithRemoteEnabled criteria would return false
+             since we do not alter the remote store based index settings at that stage. Explicitly
+             blocking checkpoint publication from this refresh listener since it ends up interfering
+             with the RemoteStoreRefreshListener invocation
+             */
+            && !shard.ongoingEngineMigration()) {
```

Comment:
can we have `isSegRepLocalEnabled` instead of negative checks ?

Reply:
Addressed this. Changed the `isSegRepLocalEnabled` check to ensure that the Checkpoint listeners won't be registered if the node is non-remote